classes/class-wc-connect-nux.php

```diff
@@ -177,15 +200,23 @@ public static function get_banner_type_to_display( $status = array() ) {
 				case self::JETPACK_ACTIVATED_NOT_CONNECTED:
 					return 'before_jetpack_connection';
 				case self::JETPACK_CONNECTED:
-					// Has the user just gone through our NUX connection flow?
-					if ( isset( $status['should_display_after_cxn_banner'] ) && $status['should_display_after_cxn_banner'] ) {
-						return 'after_jetpack_connection';
-					}
+				case self::JETPACK_DEV:
 					// Has the user already accepted our TOS? Then do nothing.
 					// Note: TOS is accepted during the after_connection banner
-					if ( isset( $status['tos_accepted'] ) && ! $status['tos_accepted'] ) {
+					if (
+						isset( $status['tos_accepted'] )
+						&& ! $status['tos_accepted']
+						&& isset( $status['can_accept_tos'] )
+						&& $status['can_accept_tos']
+					) {
 						return 'tos_only_banner';
 					}
+
+					// Has the user just gone through our NUX connection flow?
+					if ( isset( $status['should_display_after_cxn_banner'] ) && $status['should_display_after_cxn_banner'] ) {
```

Comment:
Swapped the order back and added a regression test in https://github.com/Automattic/woocommerce-services/pull/1062/commits/d8883662742f5292fa54552467bcb1f73db1d3ef